cli/ballerina-cli/src/main/java/io/ballerina/cli/task/CreateNativeImageTask.java

```diff
@@ -62,35 +61,23 @@ public void execute(Project project) {
 
         Path nativeDirectoryPath;
         String nativeImageName;
-        Target target;
-
-        try {
-            if (project.kind().equals(ProjectKind.BUILD_PROJECT)) {
-                target = new Target(project.targetDir());
-            } else {
-                target = new Target(Files.createTempDirectory("ballerina-cache" + System.nanoTime()));
-                target.setOutputPath(getExecutablePath(project));
-            }
-        } catch (IOException e) {
-            throw createLauncherException("unable to resolve target path : " + e.getMessage());
-        } catch (ProjectException e) {
-            throw createLauncherException("unable to create executable : " + e.getMessage());
-        }
-
         Path executablePath;
-        try {
-            executablePath = target.getExecutablePath(project.currentPackage()).toAbsolutePath().normalize();
-        } catch (IOException e) {
-            throw createLauncherException("unable to resolve executable path : " + e.getMessage());
-        }
 
         if (project.kind().equals(ProjectKind.SINGLE_FILE_PROJECT)) {
             Path sourceRootParent = Optional.of(project.sourceRoot().toAbsolutePath().getParent()).orElseThrow();
             nativeDirectoryPath = sourceRootParent.resolve("native");
             nativeImageName = Optional.of(sourceRootParent.getFileName()).orElseThrow().toString();
+            executablePath = getExecutablePath(project);
         } else {
-            nativeDirectoryPath = target.path().toAbsolutePath().resolve("native");
-            nativeImageName = project.currentPackage().packageName().toString();
+            Target target;
+            try {
+                target = new Target(project.targetDir());
```

Comment:
Are we creating the native image in a temporary location?

Reply:
No this section is not for the single file projects.  We are special casing the Single file project so that the naitve image is generated into a 'native' directory in the source root